### PR TITLE
Integrate AppDynamics Java Agent with CHIPS Managed Server

### PR DESCRIPTION
**Pull request by gn404**
Added script to support the optional dynamic attachment of an AppDynamics Java Agent to a CHIPS managed server running within a Docker container.

CM-1058

### CODE REVIEW

File: appdynamics/startAppDynamics.sh [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+  if [[ "${APPDYNAMICS_AGENT_ENABLED}" == true ]]
+  then
+    # Decide if env supports use based on match to naming of a higher environment
+    if [[ -n "${ENVIRONMENT_LABEL}" ]]
+    then 
+
+      # Check var contains either Live or Staging
+      if echo "$ENVIRONMENT_LABEL" | grep -q "Live\|Staging";
+      then
+        APPDYNAMICS_AGENT_ENABLED=true
```
Comment (JMetherell) [inline appdynamics/startAppDynamics.sh:188]: This line is probably redundant, as APPDYNAMICS_AGENT_ENABLED=true already at this point.

**Review by JMetherell — APPROVED** [review]
Looks great

File: appdynamics/startAppDynamics.sh (hunk 1 above)
Comment (gn404) [reply appdynamics/startAppDynamics.sh:188]: Will fix as need to cater further for where no appd zip available for local build

Comment (wpletcher) [reply appdynamics/startAppDynamics.sh:188]: Nice work!

**Review by JMetherell — APPROVED** [review]

File: appdynamics/startAppDynamics.sh [hunk 2]
```diff
@@ -204,6 +208,9 @@ f_initAppDynamicsEnabled() {
 # pre-requisites: 
 # =============================================================================
 
+# Optional env variable but needs to be defaulted if not set prior to any logging
+if [[ -z "${APPDYNAMICS_AGENT_HOME}" ]]; then APPDYNAMICS_AGENT_HOME="/apps/appdynamics/appAgent"; fi
```
Comment (JMetherell) [inline appdynamics/startAppDynamics.sh:212]: This line is fine, but an alternative way of doing that is:
APPDYNAMICS_AGENT_HOME="${APPDYNAMICS_AGENT_HOME:-/apps/appdynamics/appAgent}"

**Review by JMetherell — APPROVED** [review]

**Pull request merged**